barista/src/main/java/com/markelliot/barista/handlers/EndpointHandlerBuilder.java

```diff
@@ -16,19 +16,29 @@
 
 package com.markelliot.barista.handlers;
 
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Multimaps;
+import com.markelliot.barista.Request;
 import com.markelliot.barista.SerDe;
 import com.markelliot.barista.authz.Authz;
 import com.markelliot.barista.endpoints.EndpointHandler;
 import com.markelliot.barista.endpoints.EndpointRuntime;
 import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
 import io.undertow.server.RoutingHandler;
+import java.util.ArrayList;
+import java.util.Optional;
 import java.util.Set;
+import java.util.TreeMap;
+import java.util.function.Consumer;
 
 public final class EndpointHandlerBuilder {
+    private final Optional<Consumer<Request>> fallbackHandler;
     private final SerDe serde;
     private final Authz authz;
 
-    public EndpointHandlerBuilder(SerDe serde, Authz authz) {
+    public EndpointHandlerBuilder(Optional<Consumer<Request>> fallbackHandler, SerDe serde, Authz authz) {
```

Comment:
Since the fallbackHandler is Optional I’d prefer to put it last in the arg list rather than first.